Use the Sphinx-based document framework for the Crypto Driver Interface

athoelke opened the pull request:
This PR reworks the content of the Crypto Driver Interface to use the same Sphinx-based document building system as the other PSA APIs.

* The content is effectively identical to the original TF-PSA-Crypto document.
* Moved a small amount of introductory text to the Abstract and About sections.
* Adjusted the outline levels of a few sections to improve the overall chapter structure.
* Moved the open questions to an appendix.

This is a draft rendering of this PR: [111106-PSA_Certified_Crypto_Driver_Interface-1.0-alp.1-draft.pdf](https://github.com/user-attachments/files/22599844/111106-PSA_Certified_Crypto_Driver_Interface-1.0-alp.1-draft.pdf)

I propose to publish this as a baseline **1.0 Alpha-1** version of the document.

gilles-peskine-arm commented on doc/crypto-driver/body/entry-points.rst:
```diff
@@ -0,0 +1,954 @@
+
+The following driver entry points perform a cryptographic operation in one shot (single-part operation):
+
+*   ``"hash_compute"`` (transparent drivers only): calculation of a hash.
```
Dispatch goes to an opaque driver based on the key location. You can't reach an opaque driver for an operation with no key.

athoelke commented:
> I would already for the alpha.1 draft change the title of the document, so that the term "cryptoprocessor" disappears at least from there.

The document has always been titled "PSA Cryptoprocessor Driver Interface", although the filename has just been `psa-driver-interface.md`.

The Crypto API uses the term 'cryptoprocessor' to mean the implementation of the API, or particularly that part of the implementation that is isolated from the application. [From the Glossary](https://arm-software.github.io/psa-api/crypto/1.3/about.html#term-cryptoprocessor) in the Crypto API:

> Cryptoprocessor: The component that performs cryptographic operations. A cryptoprocessor might contain a keystore and countermeasures against a range of physical and timing attacks.

So the title is not _incorrect_, in that context. However, we could just shorten it to "[PSA] Crypto Driver Interface"? - as we did a few years ago with the Cryptography API in v1.1.1 as that was typically referred to as the Crypto API anyway.

gilles-peskine-arm commented:
Regardless of the number before alpha, I'd call it alpha 1, to allow for future checkpoints that aren't yet beta quality.

athoelke commented:
I have updated the PR with a title change to "PSA Certified Crypto Driver Interface" and kept the 1.0-alpha, but set the Issue to '1', so it is more evidently the first alpha.

I have also directed the reader to the v1.0 scope discussion post (#306) for more details about the plans for v1.0.

I have replaced the draft rendering in the PR description.
